### PR TITLE
[2213] Add support for single line format skip with other comments on the same line

### PR DESCRIPTION
**Pull request by henriholopainen**
### Description

As requested [here](https://github.com/psf/black/issues/2213), this PR implements support for single line format skip with other comments on the same line. A question for the reviewer: should this be a preview style feature?

### Checklist - did you ...

- [X] Add an entry in `CHANGES.md` if necessary?
- [X] Add / update tests if necessary?
- [X] Add new / update outdated documentation?

### CODE REVIEW

**Comment by henriholopainen** [comment]
At least [#3330](https://github.com/psf/black/issues/3330) also relates to this

File: docs/the_black_code_style/current_style.md [hunk 1]
```diff
@@ -8,12 +8,14 @@ deliberately limited and rarely added. Previous formatting is taken into account
...
-_Black_ reformats entire files in place. It doesn't reformat lines that end with
+_Black_ reformats entire files in place. It doesn't reformat lines that contain
 `# fmt: skip` or blocks that start with `# fmt: off` and end with `# fmt: on`.
-`# fmt: on/off` must be on the same level of indentation and in the same block, meaning
-no unindents beyond the initial indentation level between them. It also recognizes
-[YAPF](https://github.com/google/yapf)'s block comments to the same effect, as a
-courtesy for straddling code.
+`# fmt: skip` can be mixed with other pragmas/comments either with multiple comments
+(e.g. `# fmt: skip # pylint # noqa`) or as a semicolon separeted list (e.g.
+`# fmt: skip; pylint; noqa`). `# fmt: on/off` must be on the same level of indentation
```
Comment (JelleZijlstra) [inline docs/the_black_code_style/current_style.md:15]: I didn't see requests for this semicolon separation in the issue. Do we really need it?

**Review by JelleZijlstra — COMMENTED** [review]
Thanks!

Yes, this should be preview style only.

File: docs/the_black_code_style/current_style.md (hunk 1 above)
Comment (henriholopainen) [reply docs/the_black_code_style/current_style.md:15]: I picked it up from [this comment](https://github.com/psf/black/issues/2213#issuecomment-1411226962). I thought it was good, as I personally don't like the double comment syntax, but I'm open to drop it if you think it's better that way.

File: CHANGES.md [hunk 2]
```diff
@@ -58,6 +58,10 @@
 - Fix merging implicit multiline strings that have inline comments (#3956)
 - Allow empty first line after block open before a comment or compound statement (#3967)
 
+### Configuration
```
Comment (JelleZijlstra) [inline CHANGES.md:61]: This is in the wrong section now since we just released 23.10.1

File: docs/the_black_code_style/current_style.md (hunk 1 above)
Comment (JelleZijlstra) [reply docs/the_black_code_style/current_style.md:15]: It's probably a good idea! Let's keep it.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**